### ISSUE TITLE
When using alt buffer in bash (\e[?1049h) write writes into normal buffer

### ISSUE THREAD

**Issue by nxuv** [ssvb/speedy-stdio#1]

**Comment by ssvb** [comment]
What's the best way to reproduce this problem? The following code:

```D
void main() {
    writeln("hello 1");
    writeln("\x1B[?1049h");
    writeln("hello 2");
    writeln("\x1B[?1049l");
    writeln("hello 3");
}
```

prints

```
hello 1

hello 3
```

with both `std.stdio` and `speedy.stdio` on my computer. It would be useful to have a testcase, which can show the difference.

**Comment by nxuv** [comment]
I'll try making one, might be something else involved too

**Comment by nxuv** [comment]
```d
#!/usr/bin/env dub
/+ dub.sdl:
name "speedy"
dependency "speedy-stdio" version="~>0.1.2"
+/

static import speedy.stdio;
static import std.stdio;

void main() {
    speedy.stdio.writeln("speedy Test 1");
    std.stdio.writeln("std Test 1");

    screenEnableAltBuffer();
    
    speedy.stdio.writeln("speedy Test 2");
    std.stdio.writeln("std Test 2");

    screenDisableAltBuffer();

    speedy.stdio.writeln("speedy Test 3");
    std.stdio.writeln("std Test 3");
}

/// Enabled/Disables alternative screen buffer. 
void screenEnableAltBuffer() {
    std.stdio.write("\033[?1049h");
}
/// Ditto
void screenDisableAltBuffer() {
    std.stdio.write("\033[?1049l");
}
```

Prints:
```
std Test 1
std Test 3
speedy Test 1
speedy Test 2
speedy Test 3
```

**Comment by nxuv** [comment]
Interesting, if to use speedy for entering in alt buffer, e.g.
```d
/// Enabled/Disables alternative screen buffer. 
void screenEnableAltBuffer() {
    speedy.stdio.write("\033[?1049h");
}
/// Ditto
void screenDisableAltBuffer() {
    speedy.stdio.write("\033[?1049l");
}
```
Then it prints 
```
std Test 1
std Test 2
std Test 3
speedy Test 1
speedy Test 3
```
Instead. 

Is this supposed to be like that? Since they all are supposed to write into STDOUT_FILENO, which in most case is always 1

https://github.com/ssvb/speedy-stdio/blob/8080df680563e59b6fa8a5b460f1e0a9a9d8bea7/speedy/stdio.d#L92

**Comment by ssvb** [comment]
Thanks, now I see what's going on. The problem is that `speedy.stdio` is doing its own buffering and is not designed to be used together with `std.stdio` to access `stdout` simultaneously in an interleaved fashion. 

The function [fwrite](https://man7.org/linux/man-pages/man3/fread.3.html) has an extra buffering layer on top of [write](https://man7.org/linux/man-pages/man2/write.2.html). For example, the [fileno](https://man7.org/linux/man-pages/man3/fileno.3.html) function allows to get the file descriptor (such as `STDOUT_FILENO`) from a higher level FILE stream (such as `stdout`). I can try to experiment with writing via [stdc.stdio.fwrite](https://dlang.org/library/core/stdc/stdio/fwrite.html) without my own buffering, but a lot of performance will be lost. And this kinda defeats the purpose. I have added some notes to the project's [README](https://github.com/ssvb/speedy-stdio/blob/main/README.md) to avoid any confusion.

**Comment by nxuv** [comment]
Sad, I wanted to use it to implement my TUI library for more fps when drawing entire screen. But since this is the case I cannot use it since my general lib I'm using relies on std.stdio

You can close issue if you don't intend to make them compat

**Comment by ssvb** [comment]
This is now expected to be fixed in the v0.2.0 update.

**Comment by nxuv** [comment]
Issue resolved, closing
